### PR TITLE
fix(triggers): Pass image as an artifact when using Docker trigger

### PR DESCRIPTION
**Pull request by wjoel**
This is similar to the fix in https://github.com/spinnaker/echo/pull/219

### CODE REVIEW

File: app/scripts/modules/docker/src/pipeline/trigger/dockerTriggerOptions.directive.js [hunk 1]
```diff
@@ -67,6 +67,15 @@ module.exports = angular
 
     this.updateSelectedTag = (item) => {
       this.command.extraFields.tag = item;
+
+      const imageName = this.command.trigger.registry + '/' + this.command.trigger.repository;
+      const imageReference = imageName + ':' + item;
```
Comment (lwander) [inline app/scripts/modules/docker/src/pipeline/trigger/dockerTriggerOptions.directive.js]: Any chance `item` is a digest?

Comment (wjoel) [reply app/scripts/modules/docker/src/pipeline/trigger/dockerTriggerOptions.directive.js]: No, as far as I know. As the function name implies, the `item` here is just the selected (Docker image) tag. I'd be happy to rename the parameter to `tag` if my understanding is correct.

Comment (wjoel) [reply app/scripts/modules/docker/src/pipeline/trigger/dockerTriggerOptions.directive.js]: And the Docker image provider does not show digests at all. Only tags are available.

Comment (wjoel) [reply app/scripts/modules/docker/src/pipeline/trigger/dockerTriggerOptions.directive.js]: Done, added some checks and made `registry` optional.

File: app/scripts/modules/docker/src/pipeline/trigger/dockerTriggerOptions.directive.js [hunk 2]
```diff
@@ -65,8 +65,22 @@ module.exports = angular
...
+      this.command.extraFields.tag = tag;
+
+      if (this.command.trigger && this.command.trigger.repository) {
+        let imageName = '';
+        if (this.command.trigger.registry) {
+          imageName += this.command.trigger.registry + '/';
+        }
+        imageName += this.command.trigger.repository;
+        this.command.extraFields.artifacts = [{
+          type: "docker/image",
```
Comment (anotherchrisberry) [inline app/scripts/modules/docker/src/pipeline/trigger/dockerTriggerOptions.directive.js]: This should be in single quotes, i.e. `type: 'docker/image'`

Comment (wjoel) [reply app/scripts/modules/docker/src/pipeline/trigger/dockerTriggerOptions.directive.js]: Fixed

**Pull request merged**